### PR TITLE
fix(*): add use-sync-external-store to replace useEffect, useState

### LINKED ISSUE #566: [Feature]: use useSyncExternalStore shim than useEffect and useReducer.
### Package Scope

@suspensive/react

### Description

For DevMode, support cjs, esm. with describe extension file

### Possible Solution

_No response_

### etc.

_No response_

### PR DESCRIPTION
**Pull request by manudeli**
fix #566

# Overview

I add use-sync-external-store correctly

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/suspensive/react/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: packages/react/src/contexts/SuspensiveDevModeContext.ts [hunk 1]
```diff
@@ -1,42 +1,45 @@
-import { createContext, useContext, useEffect, useReducer } from 'react'
-import { increase } from '../utils'
+import { type ComponentProps, type ComponentType, createContext, createElement, useContext } from 'react'
+import { useSyncExternalStore } from 'use-sync-external-store/shim/index.js'
```
Comment (manudeli) [inline packages/react/src/contexts/SuspensiveDevModeContext.ts:2]: https://github.com/TanStack/query/blob/v4/packages/react-query/src/useSyncExternalStore.ts

File: packages/react/src/contexts/SuspensiveDevModeContext.spec.tsx [hunk 2]
```diff
@@ -48,4 +33,16 @@ describe('SuspensiveDevMode', () => {
...
+    const { result } = renderHook(() => useContext(DevModeContext))
+    expect(result.current).toBeNull()
+  })
+
+  it('should be get by useSuspensiveDevModeContext if SuspensiveProvider in parent', () => {
+    const { result } = renderHook(() => useContext(DevModeContext), {
+      wrapper: (props) => <SuspensiveProvider {...props} value={new Suspensive()} />,
+    })
+    expect(result.current).toBeInstanceOf(SuspensiveDevMode)
+  })
```
Comment (tooooo1) [inline packages/react/src/contexts/SuspensiveDevModeContext.spec.tsx:47]: How about using `describe`? for instance:

```suggestion
  describe("DevModeContext", () => {
    it("returns null when no SuspensiveProvider is present", () => {
      const { result } = renderHook(() => useContext(DevModeContext))
      expect(result.current).toBeNull()
    })

    it("returns an instance of SuspensiveDevMode when wrapped with SuspensiveProvider", () => {
      const { result } = renderHook(() => useContext(DevModeContext), {
        wrapper: (props) => (
          <SuspensiveProvider {...props} value={new Suspensive()} />
        ),
      })
      expect(result.current).toBeInstanceOf(SuspensiveDevMode)
    })
  })
```

Comment (manudeli) [reply packages/react/src/contexts/SuspensiveDevModeContext.spec.tsx:47]: Cool

**Review by tooooo1 — APPROVED** [review]
good

**Comment by manudeli** [comment]
@minsoo-web @okinawaa Realert you guys

**Review by minsoo-web — APPROVED** [review]
Sorry for the late check, many thanks!

**Pull request merged**